### PR TITLE
Suggest a default database name from instanceId + database.

### PR DESCRIPTION
**Pull request by yuchenshi**
Tested

image

### CODE REVIEW

**Review by joehan — APPROVED** [review]
LGTM, very nice defensive coding here to have a safe fallback

**Comment by yuchenshi** [comment]
> LGTM, very nice defensive coding here to have a safe fallback

I?.have?.no?.idea?.what?.you?.are?.talking || 'about'

**Comment by joehan** [comment]
> > LGTM, very nice defensive coding here to have a safe fallback
> 
> I?.have?.no?.idea?.what?.you?.are?.talking || 'about'

Suggest:
```
 I?.have?.no?.idea?.what?.you?.are?.talking ?? 'about'
```

File: firebase-vscode/src/data-connect/emulator.ts [hunk 1]
```diff
@@ -48,9 +50,26 @@ export class DataConnectEmulatorController implements vscode.Disposable {
...
   private async connectToPostgres() {
     const rc = firebaseRC.value?.tryReadValue;
+    let localConnectionString =
+      rc?.getDataconnect()?.postgres?.localConnectionString;
+    if (!localConnectionString) {
+      const dataConnectConfigsValue =
+        await firstWhereDefined(dataConnectConfigs);
+      let dbname = "postgres";
+      const postgresql =
+        dataConnectConfigsValue?.tryReadValue.values[0]?.value?.schema
```
Comment (hlshen) [inline firebase-vscode/src/data-connect/emulator.ts:60]: we can make this available in the ResolveDataConnectConfig class.

Comment (yuchenshi) [reply firebase-vscode/src/data-connect/emulator.ts:60]: I'll clean this up after the release :)

**Review by hlshen — APPROVED** [review]

File: firebase-vscode/src/data-connect/emulator.ts (hunk 1 above)
Comment (hlshen) [reply firebase-vscode/src/data-connect/emulator.ts:60]: i can do it :) much clean up to be done

Comment (yuchenshi) [reply firebase-vscode/src/data-connect/emulator.ts:60]: Sure, I'll leave it up to you then

**Pull request merged**